autogpt_platform/backend/backend/util/virus_scanner.py

```diff
@@ -115,11 +122,19 @@ async def scan_file(
                     f"Scanning {filename} with chunk size: {chunk_size // 1_048_576}MB"
                 )
 
-                # Scan all chunks with current chunk size
-                for offset in range(0, len(content), chunk_size):
-                    chunk_data = content[offset : offset + chunk_size]
-                    infected, threat = await self._scan_chunk(chunk_data)
+                # Fire off all chunk scans concurrently
+                tasks = [
+                    asyncio.create_task(
+                        self._scan_chunk(content[offset : offset + chunk_size])
+                    )
+                    for offset in range(0, len(content), chunk_size)
+                ]
+
+                for coro in asyncio.as_completed(tasks):
+                    infected, threat = await coro
                     if infected:
+                        for t in tasks:
+                            t.cancel()
```

Comment:
the cancellation can be prefixed with `if not t.done()`

Reply:
Good spot! updated